gensim/models/wrappers/ldavowpalwabbit.py

```diff
@@ -0,0 +1,537 @@
+
+"""
+Python wrapper around Vowpal Wabbit's Latent Dirichlet Allocation (LDA)
+implementation [1]_.
+
+This uses Matt Hoffman's online algorithm, for LDA [2]_, i.e. the same
+algorithm that Gensim's LdaModel is based on.
+
+Note: Currently working and tested with Vowpal Wabbit 7.10. Vowpal Wabbit's
+API isn't currently stable, so this may or may not work with older/newer
+versions. The aim will be to ensure this wrapper always works with the latest
+release of Vowpal Wabbit.
+
+Tested with python 2.6, 2.7, and 3.4.
+
+Example:
+
+    Train model:
+    >>> lda = gensim.models.wrappers.LdaVowpalWabbit('/usr/local/bin/vw',
+                                                     corpus=corpus,
+                                                     num_topics=20,
+                                                     id2word=dictionary)
+
+    Update existing model:
+    >>> lda.update(another_corpus)
+
+    Get topic probability distributions for a document:
+    >>> print(lda[doc_bow])
+
+    Print 10 topics:
+    >>> print(lda.print_topics())
+
+    Save/load the trained model:
+    >>> lda.save('vw_lda.model')
+    >>> lda = gensim.models.wrappers.LdaVowpalWabbit.load('vw_lda.model')
+
+    Get bound on log perplexity for given test set:
+    >>> print(lda.log_perpexity(test_corpus))
+
+Vowpal Wabbit works on files, so this wrapper maintains a temporary directory
+while it's around, reading/writing there as necessary.
+
+Output from Vowpal Wabbit is logged at either INFO or DEBUG levels, enable
+logging to view this.
+
+.. [1] https://github.com/JohnLangford/vowpal_wabbit/wiki
+.. [2] http://www.cs.princeton.edu/~mdhoffma/
+"""
+
+from __future__ import unicode_literals
+from __future__ import print_function
+from __future__ import division
+
+import os
+import logging
+import tempfile
+import shutil
+import subprocess
+
+import numpy
+
+from gensim import utils
+
+LOG = logging.getLogger('gensim.models.ldavowpalwabbit')
+
+
+class LdaVowpalWabbit(utils.SaveLoad):
+    """Class for LDA training using Vowpal Wabbit's online LDA. Communication
+    between Vowpal Wabbit and Python takes place by passing around data files
+    on disk and calling the 'vw' binary with the subprocess module.
+    """
+    def __init__(self, vw_path, corpus=None, num_topics=100, id2word=None,
+                 chunksize=256, passes=1, alpha=0.1, eta=0.1, decay=0.5,
+                 offset=1, gamma_threshold=0.001, random_seed=None,
+                 cleanup_files=True, tmp_prefix='tmp'):
+        """`vw_path` is the path to Vowpal Wabbit's 'vw' executable.
+
+        `corpus` is an iterable training corpus. If given, training will
+        start immediately, otherwise the model is left untrained (presumably
+        because you want to call `update()` manually).
+
+        `num_topics` is the number of requested latent topics to be extracted
+        from the training corpus.
+        Corresponds to VW's '--lda <num_topics>' argument.
```

Comment:
Thanks so much for adding these lines to the documentation. Makes it very clear how the parameters translate.